commands/miner.go

```diff
@@ -46,6 +46,9 @@ var minerPledgeCmd = &cmds.Command{
 		if err != nil {
 			return err
 		}
+		if minerAddr.Protocol() != address.Actor {
```

Comment:
This isn't right.

We have 4 different types of addresses:
1. ID, which we don't really use
2. Actor, which we use when we we aren't generating the address from a public key
3. Secp256k, which we use when we do generate the address from a public key
4. BLS, which we don't use yet

In theory all the addresses are interchangeable in the sense that any actor could be addressed by any type of address.

In practice, AccountActors have addresses that represent their associated public keys and all other actors are generated from a string (address.Actor protocol). It's probably true that a MinerActor will never be addressed by a Secp256k address, but it's not true that every Actor address points to a miner, or that MinerActor's will always be addressed by an Actor address.

If we want more intuitive error messages from this command, we really want to get the actor from latest state (`ActorGet`) and assert that its code Cid equals `types.MinerActorCodeCid`.